src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import dao.FileReaderImpl;
+import dao.FileWriter;
+import dao.FileWriterImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import model.OperationType;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import service.FruitShopService;
+import strategy.BalanceOperationHandler;
+import strategy.OperationHandler;
+import strategy.PurchaseOperationHandler;
+import strategy.ReturnOperationHandler;
+import strategy.SupplyOperationHandler;
+
+public class Main {
+    private static final Logger logger = LogManager.getLogger(Main.class);
+
+    public static void main(String[] args) {
+        if (args.length < 2) {
+            logger.error("Please provide both source and target file paths as arguments.");
+            return;
+        }
+
+        final String sourceFilePath = args[0];
+        final dao.FileReader fileReader = new FileReaderImpl();
```

Comment:
Don't hardcode the type `dao.FileReader` on the right side of the assignment. Use the interface type on the left side to ensure that you can easily switch implementations if needed.